python/tests/expressions/test_visitors.py

```diff
@@ -874,57 +872,32 @@ def test_manifest_evaluator_less_than_overlap():
             upper_bound=_to_byte_buffer(StringType(), "b"),
         )
     )
+    assert not _ManifestEvalVisitor(string_schema, expr).eval(manifest)
 
-    assert not _create_manifest_evaluator(expr).eval(manifest)
-
-
-def test_manifest_evaluator_less_than_all_null():
-    expr = BoundLessThan(
-        term=BoundReference(
-            field=NestedField(field_id=1, name="foo", field_type=StringType(), required=False),
-            accessor=Accessor(position=0, inner=None),
-        ),
-        literal=StringLiteral("a"),
-    )
 
+def test_manifest_evaluator_less_than_all_null(string_schema: Schema):
```

Comment:
You can probably make this generic and test that all inequalities will reject a column of all nulls. The only predicates that should accept this metadata are `!=` and `NOT IN`.